src/codec.rs

```diff
@@ -1099,13 +1099,32 @@ impl Decode for bool {
 	}
 }
 
+impl Encode for Duration {
+	fn size_hint(&self) -> usize {
+		mem::size_of::<u64>() + mem::size_of::<u32>()
+	}
+
+	fn encode(&self) -> Vec<u8> {
+		let secs = self.as_secs();
+		let nanos = self.subsec_nanos();
+		(secs, nanos).encode()
+	}
+}
+
+impl Decode for Duration {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		let (secs, nanos) = <(u64, u32)>::decode(input)?;
+		Ok(Duration::from_secs(secs) + Duration::from_nanos(nanos as u64))
```

Comment:
I think would be better to check that nanos is less than a 10^9 in order to make the encoding unique.

(Also if you check this then you can directly use Duration::new)

Reply:
Good point!